client/client.go

```diff
@@ -2933,43 +2929,27 @@ DISCOLOOP:
 		}
 
 		for _, s := range consulServices {
-			port := strconv.Itoa(s.ServicePort)
-			addrstr := s.ServiceAddress
-			if addrstr == "" {
-				addrstr = s.Address
-			}
-			addr, err := net.ResolveTCPAddr("tcp", net.JoinHostPort(addrstr, port))
-			if err != nil {
-				mErr.Errors = append(mErr.Errors, err)
-				continue
-			}
-
-			// Query the members from the region that Consul gave us, and
-			// extract the client-advertise RPC address from each member
-			var membersResp structs.ServerMembersResponse
-			if err := c.connPool.RPC(region, addr, "Status.Members", rpcargs, &membersResp); err != nil {
-				mErr.Errors = append(mErr.Errors, err)
-				continue
-			}
-			for _, member := range membersResp.Members {
-				if addrTag, ok := member.Tags["rpc_addr"]; ok {
-					if portTag, ok := member.Tags["port"]; ok {
-						addr, err := net.ResolveTCPAddr("tcp",
-							fmt.Sprintf("%s:%s", addrTag, portTag))
-						if err != nil {
-							mErr.Errors = append(mErr.Errors, err)
-							continue
-						}
-						srv := &servers.Server{Addr: addr}
-						nomadServers = append(nomadServers, srv)
-					}
+			if slices.Contains(s.ServiceTags, region) {
```

Comment:
I've updated the PR (and the description above) to drop the extra tags/meta on Consul. And then I've updated the `registerNode` method to use the servers we get back from the response. If we get no servers or a "no path to region", we'll bubble that state up to the retry loop so that it triggers discovery again and hits the fast-path retry rather than the longer retry of 15s.